fedimint-api/src/core.rs

```diff
@@ -134,44 +185,25 @@ macro_rules! module_plugin_trait_define {
 #[macro_export]
 macro_rules! plugin_types_trait_impl {
     ($key:expr, $input:ty, $output:ty, $outcome:ty, $ci:ty, $cache:ty) => {
-        impl fedimint_api::core::PluginInput for $input {
-            fn module_key(&self) -> ModuleKey {
-                $key
-            }
-        }
+        impl fedimint_api::core::PluginInput for $input {}
 
-        impl fedimint_api::core::PluginOutput for $output {
-            fn module_key(&self) -> ModuleKey {
-                $key
-            }
-        }
+        impl fedimint_api::core::PluginOutput for $output {}
 
-        impl fedimint_api::core::PluginOutputOutcome for $outcome {
-            fn module_key(&self) -> ModuleKey {
-                $key
-            }
-        }
+        impl fedimint_api::core::PluginOutputOutcome for $outcome {}
 
-        impl fedimint_api::core::PluginConsensusItem for $ci {
-            fn module_key(&self) -> ModuleKey {
-                $key
-            }
-        }
+        impl fedimint_api::core::PluginConsensusItem for $ci {}
 
-        impl fedimint_api::server::PluginVerificationCache for $cache {
-            fn module_key(&self) -> ModuleKey {
-                $key
-            }
-        }
+        impl fedimint_api::server::PluginVerificationCache for $cache {}
     };
 }
 
 macro_rules! erased_eq {
     ($newtype:ty) => {
         fn erased_eq(&self, other: &$newtype) -> bool {
-            if self.module_key() != other.module_key() {
-                return false;
-            }
+            // TODO:?
+            // if self.module_key() != other.module_key() {
+            //     return false;
+            // }
```

Comment:
Needs resolution, if we can guarantee equality of instance id on another level we can delete it here, otherwise we need to figure out how to have access to it here.